### PR TITLE
docs: surface repository_url for maven

### PR DESCRIPTION
**Pull request by andrewpollock**
Make it easier for casual readers focusing on Maven to become aware of the `repository_url` qualifier

### CODE REVIEW

File: PURL-TYPES.rst [hunk 1]
```diff
@@ -402,6 +402,7 @@ maven
       pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?type=zip&classifier=dist
       pkg:maven/net.sf.jacob-projec/jacob@1.14.3?classifier=x86&type=dll
       pkg:maven/net.sf.jacob-projec/jacob@1.14.3?classifier=x64&type=dll
+      pkg:maven/groovy/groovy@1.0?repository=https://repo1.maven.org/maven2
```
Comment (matt-phylum) [inline PURL-TYPES.rst:405]:
```suggestion
      pkg:maven/groovy/groovy@1.0?repository_url=https://repo1.maven.org/maven2
```

Comment (andrewpollock) [reply PURL-TYPES.rst:405]: Thank you

**Review by matt-phylum — APPROVED** [review]

File: PURL-TYPES.rst [hunk 2]
```diff
@@ -402,6 +402,7 @@ maven
       pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?type=zip&classifier=dist
       pkg:maven/net.sf.jacob-projec/jacob@1.14.3?classifier=x86&type=dll
       pkg:maven/net.sf.jacob-projec/jacob@1.14.3?classifier=x64&type=dll
+      pkg:maven/groovy/groovy@1.0?repository_url=https://repo1.maven.org/maven2
```
Comment (pombredanne) [inline PURL-TYPES.rst:405]: Could we have another repo for the example? I do not want to promote that adding the default is a thing. 
May be instead something like: `pkg:maven/groovy/groovy@1.0?repository_url=https://maven.google.com`

Comment (andrewpollock) [reply PURL-TYPES.rst]: Good call. Addressed in [c131917](https://github.com/package-url/purl-spec/pull/322/commits/c131917b41bd6c492a216a31982cb8a5ce09b586)

**Review by pombredanne — APPROVED** [review]
Thanks! merging.

**Pull request merged**